doc/source/rllib/rllib-checkpoints-and-exports.rst

```diff
@@ -0,0 +1,346 @@
+
+We have already looked at the ``policies/`` sub-directory inside an
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm` checkpoint dir
+and learned that individual policies inside the
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm` store all their state
+information under their policy ID inside that sub-directory.
+Thus, we now have the entire picture of a checkpoint:
+
+.. code-block::
+
+    .
+    ..
+    .is_checkpoint
+    .tune_metadata
+
+    rllib_checkpoint.json           # <- checkpoint info, such as checkpoint version, e.g. "1.0"
+    algorithm_state.pkl             # <- state of the Algorithm (excluding Policy states)
+
+    policies/
+
+        policy_A/
+            rllib_checkpoint.json   # <- checkpoint info, such as checkpoint version, e.g. "1.0"
+            policy_state.pkl        # <- state of policy_A
+
+        policy_B/
+            rllib_checkpoint.json   # <- checkpoint info, such as checkpoint version, e.g. "1.0"
+            policy_state.pkl        # <- state of policy_B
+
+
+How do I create a Policy checkpoint?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create a :py:class:`~ray.rllib.policy.policy.Policy` checkpoint by
+either calling ``save()`` on your :py:class:`~ray.rllib.algorithms.algorithm.Algorithm`, which
+will save each individual Policy's checkpoint under the ``policies/`` sub-directory as
+described above or - if you need more fine-grained control - by doing the following:
+
+
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
+    :start-after: __create-policy-checkpoint-begin__
+    :end-before: __create-policy-checkpoint-end__
+
+If you now check out the provided directory (``/tmp/my_policy_checkpoint/``), you
+should see the following files in there:
+
+.. code-block::
+
+    .
+    ..
+    rllib_checkpoint.json   # <- checkpoint info, such as checkpoint version, e.g. "1.0"
+    policy_state.pkl        # <- state of "pol1"
+
+
+How do I restore from a Policy checkpoint?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Assume you would like to serve your trained policy(ies) in production and would therefore
+like to use only the RLlib :py:class:`~ray.rllib.policy.policy.Policy` instance,
+without all the other functionality that
+normally comes with the :py:class:`~ray.rllib.algorithms.algorithm.Algorithm` object, like different ``RolloutWorkers`` for collecting
+training samples or for evaluation (both of which include RL environment copies), etc..
+
+In this case, it would be quite useful if you had a way to restore just the
+:py:class:`~ray.rllib.policy.policy.Policy`
+from either a :py:class:`~ray.rllib.policy.policy.Policy` checkpoint or an
+:py:class:`~ray.rllib.algorithms.algorithm.Algorithm` checkpoint, which - as we learned above -
+contains all its Policies' checkpoints.
+
+Here is how you can do this:
+
+.. literalinclude:: ../../../rllib/examples/documentation/checkpoints_and_exports.py
+    :language: python
```

Comment:
not really your problem for this PR (but maybe you have an idea!), but it isn't great that code comments like `#noqa` leak into the docs:

![Screenshot 2022-10-07 at 13 20 09](https://user-images.githubusercontent.com/3462566/194541909-26a475b3-111c-43b8-b185-d32d86e06f73.png)

Reply:
done